src/cmd/list.go

```diff
@@ -226,35 +266,65 @@ func listOutput(images, containers []map[string]interface{}) {
 			"STATUS",
 			"IMAGE NAME")
 
-		var idKey, createdKey, statusKey string
-		if podman.CheckVersion("2.0.0") {
-			idKey = "Id"
-			createdKey = "CreatedAt"
-			statusKey = "State"
-		} else {
-			idKey = "ID"
-			createdKey = "Created"
-			statusKey = "Status"
+		for _, container := range containers {
+			var id, name, created, status, imageName string
+			id = utils.ShortID(container.ID)
+			name = container.GetName()
+			status = container.GetStatus()
+			created = container.GetCreatedString()
+			imageName = container.Image
+
+			fmt.Fprintf(writer, "%s\t%s\t%s\t%s\t%s\n", id, name, created, status, imageName)
 		}
 
-		for _, container := range containers {
-			id := utils.ShortID(container[idKey].(string))
+		writer.Flush()
+	}
+}
 
-			var nameString string
-			switch name := container["Names"].(type) {
-			case string:
-				nameString = name
-			case []interface{}:
-				nameString = name[0].(string)
-			}
+// GetCreatedString returns a string with human-readable time when an image was
+// created
+func (i toolboxImage) GetCreatedString() string {
```

Comment:
Using getters is definitely an OK approach. I'd probably think about doing something with a custom unmarshal function

``` go
func (t *toolboxImage) UnmarshalJSON(data []byte) error {     
    var raw struct {
	ID      string
	Name    string
	Names   []string
	Status  string
	State   string
	Created interface{}
	Image   string
    }

   if err := json.Unmarshal(data, &raw); err != nil {       
       return err     
    }   

   t.ID = raw.ID
   if raw.Name != "" {
      t.name = c.Name
   } else {
      t.name = c.Names[0]
   }
   [...]
```

This puts all the JSON compatibility hacks into one place - with the getter approach, code that is using toolboxImage has to be aware that some fields are OK to access, and some fields need to use getters to sanitize them.

Reply:
Again, you're right. This sounds easier to maintain.

Reply:
The latest iteration of the PR uses this approach.